src/main/java/uk/gov/pay/api/model/PaymentEvents.java

```diff
@@ -23,9 +24,11 @@
 
     public static PaymentEvents createPaymentEventsResponse(JsonNode payload, String paymentLink) {
         List<PaymentEvent> events = newArrayList();
-        if(payload.get("events").isArray()) {
-            for (JsonNode event : payload.get("events")) {
-                events.add(createPaymentEvent(event, paymentLink));
+        String paymentId = payload.get("charge_id").asText();
```

Comment:
Wouldn't this still throw a NPE if there's no `charge_id` in the response from the connector?

Also is it legal to send back some events without paymentIds?

How about some unit tests to cover this?